### ISSUE TITLE
Loss function does not decrease

### ISSUE THREAD

**Issue by AugusTue** [rohitgajawada/Where-are-they-looking-PyTorch#1]
The loss remains at 2.97, which is close to the initial value，its performance is as follows:
`Epoch: [3][1700/2512]   Time 0.556 (946.500)    Data 0.546 (929.288)    Loss 2.977
Epoch: [3][1750/2512]   Time 0.557 (975.468)    Data 0.547 (957.740)    Loss 2.977
Epoch: [3][1800/2512]   Time 0.557 (1002.367)   Data 0.546 (984.131)    Loss 2.976
Epoch: [3][1850/2512]   Time 0.556 (1029.715)   Data 0.546 (1010.939)   Loss 2.976
Epoch: [3][1900/2512]   Time 0.557 (1058.871)   Data 0.547 (1039.615)   Loss 2.975
Epoch: [3][1950/2512]   Time 0.556 (1085.180)   Data 0.546 (1065.417)   Loss 2.975
Epoch: [3][2000/2512]   Time 0.556 (1112.911)   Data 0.546 (1092.618)   Loss 2.976
Epoch: [3][2050/2512]   Time 0.557 (1141.840)   Data 0.547 (1121.041)   Loss 2.976
Epoch: [3][2100/2512]   Time 0.556 (1168.497)   Data 0.546 (1147.199)   Loss 2.975
Epoch: [3][2150/2512]   Time 0.556 (1196.627)   Data 0.546 (1174.804)   Loss 2.975
Epoch: [3][2200/2512]   Time 0.557 (1225.442)   Data 0.547 (1203.099)   Loss 2.975
Epoch: [3][2250/2512]   Time 0.556 (1252.026)   Data 0.546 (1229.191)   Loss 2.975
Epoch: [3][2300/2512]   Time 0.556 (1279.078)   Data 0.546 (1255.758)   Loss 2.975
Epoch: [3][2350/2512]   Time 0.556 (1308.258)   Data 0.546 (1284.438)   Loss 2.975
Epoch: [3][2400/2512]   Time 0.556 (1334.303)   Data 0.546 (1310.024)   Loss 2.976
Epoch: [3][2450/2512]   Time 0.555 (1360.843)   Data 0.545 (1336.055)   Loss 2.975
Epoch: [3][2500/2512]   Time 0.556 (1390.264)   Data 0.546 (1364.993)   Loss 2.976
Train: [3]      Time 1393.267   Data 1367.887   Loss 2.976
Epoch: [3][0/38]        Time 7.631 (7.631)      Data 7.345 (7.345)      Dist 0.564      MinDist 0.393   
Val: [3]        Time 48.641     Data 38.395     Dist 0.572      MinDist 0.415   
`
I am not sure if it is related to these warnings when I running this project. 

1.  =>no checkpoint found at 'none'
/usr/local/lib/python2.7/dist-packages/skimage/transform/_warps.py:84:` UserWarning: The default mode, 'constant', will be changed to 'reflect' in skimage 0.15.
  warn("The default mode, 'constant', will be changed to 'reflect' in "
('Starting epoch number:', 1, 'Learning rate:', 0.001)
/Where-are-they-looking-PyTorch-master/train.py:46: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  self.losses.update(loss.data[0], inputs_size)

2.  These SourceChangeWarnings:
"/usr/local/lib/python2.7/dist-packages/torch/serialization.py:367: SourceChangeWarning: source code of class 'torch.nn.modules.container.Sequential' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/usr/local/lib/python2.7/dist-packages/torch/serialization.py:336: UnicodeWarning: Unicode unequal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if original_source != current_source:
/usr/local/lib/python2.7/dist-packages/torch/serialization.py:367: SourceChangeWarning: source code of class 'torch.nn.modules.conv.Conv2d' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/usr/local/lib/python2.7/dist-packages/torch/serialization.py:367: SourceChangeWarning: source code of class 'torch.nn.modules.activation.ReLU' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/usr/local/lib/python2.7/dist-packages/torch/serialization.py:367: SourceChangeWarning: source code of class 'torch.nn.modules.pooling.MaxPool2d' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/usr/local/lib/python2.7/dist-packages/torch/serialization.py:367: SourceChangeWarning: source code of class 'torch.nn.modules.dropout.Dropout' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/usr/local/lib/python2.7/dist-packages/torch/serialization.py:367: SourceChangeWarning: source code of class 'torch.nn.modules.linear.Linear' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)"

I hope you can help solve this problem. Looking forward to your reply. Thanks.

**Comment by rohitgajawada** [comment]
Hi MengsiYan, I have recently realized there are several bugs with our implementation after going through the paper again. We plan to address these soon and try to attain the results similar to the paper. The bugs are related to the shifted grids and label size part. Also, we did not perform data augmentation and we should.

If you require urgently, you can switch off the shifted grids and make the number of labels to be 100 and output fc layer to be 10x10 which would give you some okayish results but not perfect

**Comment by rohitgajawada** [comment]
Hi, the issue have been fixed mostly and the model is working decently. Please feel free to take a look :)
